sat_vapor_pres/include/sat_vapor_pres_k.inc

```diff
@@ -1128,893 +1009,1202 @@
+                      + real(2.0_r8_kind,FMS_SVP_KIND_)*del*real(D2TABLE2(ind+1),FMS_SVP_KIND_)
      endif
    enddo
    enddo
    enddo
 
- end subroutine lookup_es2_des2_k_3d
+ end subroutine LOOKUP_ES2_DES2_K_3D_
 
 !#######################################################################
 
- subroutine lookup_es2_des2_k_2d (temp, esat, desat, nbad)
- real, intent(in),  dimension(:,:)  :: temp
- real, intent(out), dimension(:,:)  :: esat, desat
+ subroutine LOOKUP_ES2_DES2_K_2D_ (temp, esat, desat, nbad)
+ real(kind=FMS_SVP_KIND_), intent(in),  dimension(:,:)  :: temp
+ real(kind=FMS_SVP_KIND_), intent(out), dimension(:,:)  :: esat, desat
  integer, intent(out)               :: nbad
 
- real    :: tmp, del
+ real(kind=FMS_SVP_KIND_) :: tmp, del
  integer :: ind, i, j
 
+ real(kind=FMS_SVP_KIND_) :: dtres_loc, tepsl_loc, tminl_loc,  dtinvl_loc
+
+   dtres_loc=real(dtres, FMS_SVP_KIND_)
+   tminl_loc=real(tminl, FMS_SVP_KIND_)
+   tepsl_loc=real(tepsl, FMS_SVP_KIND_)
+   dtinvl_loc=real(dtinvl, FMS_SVP_KIND_)
+
    nbad = 0
    do j = 1, size(temp,2)
    do i = 1, size(temp,1)
-     tmp = temp(i,j)-tminl
-     ind = int(dtinvl*(tmp+tepsl))
+     tmp = temp(i,j)-tminl_loc
+     ind = int(dtinvl_loc*(tmp+tepsl_loc))
      if (ind < 0 .or. ind >= table_siz)  then
        nbad = nbad+1
      else
-       del = tmp-dtres*real(ind)
-       esat(i,j) = TABLE2(ind+1) + &
-                   del*(DTABLE2(ind+1) + del*D2TABLE2(ind+1))
-       desat(i,j) = DTABLE2(ind+1) + 2.*del*D2TABLE2(ind+1)
+       del = tmp-dtres_loc*real(ind,FMS_SVP_KIND_)
+       esat(i,j) = real(TABLE2(ind+1),FMS_SVP_KIND_) &
+                   + del*( real(DTABLE2(ind+1),FMS_SVP_KIND_) &
+                                 + del*real(D2TABLE2(ind+1),FMS_SVP_KIND_) )
+       desat(i,j) = real(DTABLE2(ind+1),FMS_SVP_KIND_) &
+                    + real(2.0_r8_kind,FMS_SVP_KIND_)*del*real(D2TABLE2(ind+1),FMS_SVP_KIND_)
      endif
    enddo
    enddo
 
- end subroutine lookup_es2_des2_k_2d
+ end subroutine LOOKUP_ES2_DES2_K_2D_
 
 !#######################################################################
 
- subroutine lookup_es2_des2_k_1d (temp, esat, desat, nbad)
- real, intent(in),  dimension(:)  :: temp
- real, intent(out), dimension(:)  :: esat, desat
+ subroutine LOOKUP_ES2_DES2_K_1D_ (temp, esat, desat, nbad)
+ real(kind=FMS_SVP_KIND_), intent(in),  dimension(:)  :: temp
+ real(kind=FMS_SVP_KIND_), intent(out), dimension(:)  :: esat, desat
  integer, intent(out)             :: nbad
 
- real    :: tmp, del
+ real(kind=FMS_SVP_KIND_) :: tmp, del
  integer :: ind, i
 
+ real(kind=FMS_SVP_KIND_) :: dtres_loc, tepsl_loc, tminl_loc,  dtinvl_loc
+
+   dtres_loc=real(dtres, FMS_SVP_KIND_)
+   tminl_loc=real(tminl, FMS_SVP_KIND_)
+   tepsl_loc=real(tepsl, FMS_SVP_KIND_)
+   dtinvl_loc=real(dtinvl, FMS_SVP_KIND_)
+
    nbad = 0
    do i = 1, size(temp,1)
-     tmp = temp(i)-tminl
-     ind = int(dtinvl*(tmp+tepsl))
+     tmp = temp(i)-tminl_loc
+     IND = INT(dtinvl_loc*(tmp+tepsl_loc))
```

Comment:
This is randomly capitialized.  It doesn't fit the style of the rest of the module